### PR TITLE
Added QR scanner to wallet send card

### PR DESCRIPTION
**Pull request by soup-jingle**
## PR Checklist

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [ ] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: #202 

## What is the current behavior?
There is no QR scanner for sending funds to an address encoded as a QR code.

## What is the new behavior?
In the Wallet's Send card, there is now QR scanning functionality which will fill the recipient address with the value the scanner gets.

## Other information
The QR library (zxing 4.1.0) works on SDK 24 and up, but I added a 'fallback' for 3.3.0 as they advise so the minimum SDK of 21 should still work and be able to use the functionality. I was not, however, able to test this on any devices with that SDK version.

### CODE REVIEW

**Review by kekkyojin — COMMENTED** [review]
Not a formal review, but these code style changes would be required

File: app/build.gradle [hunk 1]
```diff
@@ -134,7 +134,10 @@ dependencies {
...
     __32bitImplementation 'io.lbry:lbrysdk32:0.94.1'
     __64bitImplementation 'io.lbry:lbrysdk64:0.94.1'
+
+    implementation ('com.journeyapps:zxing-android-embedded:4.1.0') { transitive = false }
+    implementation 'com.google.zxing:core:3.3.0'
 }
 
 apply plugin: 'com.google.gms.google-services'
-GoogleServicesPlugin.config.disableVersionCheck = true
+GoogleServicesPlugin.config.disableVersionCheck = true
```
Comment (kekkyojin) [inline app/build.gradle:143]: Revert this change

File: app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java [hunk 2]
```diff
@@ -715,8 +727,8 @@ public void onWalletBalanceUpdated(WalletBalance walletBalance) {
         } else {
             firstDatasetNotification = false;
             if (!detailRows.get(0).detailAmount.equals(earnedBalance.detailAmount)
-                 || detailRows.get(0).isInProgress != earnedBalance.isInProgress
-                 || detailRows.get(0).isUnlockable != earnedBalance.isUnlockable) {
+                    || detailRows.get(0).isInProgress != earnedBalance.isInProgress
+                    || detailRows.get(0).isUnlockable != earnedBalance.isUnlockable) {
```
Comment (kekkyojin) [inline app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java:731]: Revert the changes on these lines

File: app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java [hunk 3]
```diff
@@ -800,4 +812,13 @@ private void checkRewardsDriver() {
...
+
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == IntentIntegrator.REQUEST_CODE) {
+            IntentResult intentResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, data);
+            String code = intentResult.getContents();
+            inputSendAddress.setText(code);
+        }
+    }
+
+}
```
Comment (kekkyojin) [inline app/src/main/java/io/lbry/browser/ui/wallet/WalletFragment.java:824]: Revert the change on this last line 824

**Comment by tzarebczan** [comment]
Thanks so much for the PR! We'll get another closer review on it soon. 

Can we show you some [appreciation](https://lbry.com/faq/appreciation)?

File: app/src/main/res/drawable-anydpi/ic_qr_scanner.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24"
+    android:tint="?attr/colorControlNormal">
+  <path
+      android:fillColor="@android:color/white"
```
[the remark this answers is not in the document]
Comment (kekkyojin) [reply app/src/main/res/drawable-anydpi/ic_qr_scanner.xml:8]: I have tested it on Android 8.1. That API Level doesn't support dark mode, but enabling dark mode on app settings does the same.

Button image is still visible, so this conversation can be resolved.

**Review by kekkyojin — APPROVED** [review]
I have tested this. It works as expected. I does decode a LBRY Credits address and writes it on the expected field.

In order to test this, it was needed to merge changes from master branch. The parent commit PR author branched from was using a repository for AAR which is no longer working. That was fixed on master. After merging master into PR branch, this will build without a problem.

**Review by akinwale — APPROVED** [review]
